net/nimble/controller/src/ble_ll_adv.c

```diff
@@ -1230,11 +1282,8 @@ ble_ll_adv_sm_stop(struct ble_ll_adv_sm *advsm)
     if (advsm->adv_enabled) {
         /* Remove any scheduled advertising items */
         ble_ll_sched_rmv_elem(&advsm->adv_sch);
-
 #if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_EXT_ADV)
-        if (!(advsm->props & BLE_HCI_LE_SET_EXT_ADV_PROP_LEGACY)) {
-            ble_ll_sched_rmv_elem(&advsm->adv_secondary_sch);
-        }
+        ble_ll_sched_rmv_elem(&advsm->adv_secondary_sch);
```

Comment:
isn't this check needed?

Reply:
for legacy we won't have anything scheduled on aux so this call won't do anything - the result will be the same with and without flag so just removed it since I use similar "logic" in other places